Azure.ResourceManager.Core/Resources/LocationContainer.cs

```diff
@@ -0,0 +1,111 @@
+        /// </summary>
+        public SubscriptionOperations DefaultSubscription { get; private set; }
+
+        /// <summary>
+        /// Gets default subscription.
+        /// </summary>
+        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="P:System.Threading.CancellationToken.None" />. </param>
+        /// <returns> A <see cref="Task"/> that on completion returns the subscription id. </returns>
+        internal async Task<string> GetDefaultSubscription(CancellationToken cancellationToken = default)
+        {
+            string sub = DefaultSubscription?.Id?.Subscription;
+            if (null == sub)
+            {
+                sub = await GetSubscriptionContainer().GetDefaultSubscriptionAsync(cancellationToken);
+            }
+
+            return sub;
+        }
+
+        /// <summary>
+        /// Lists all geo-locations.
+        /// </summary>
+        /// <param name="subscriptionId"> The Id of the target subscription. </param>
+        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="P:System.Threading.CancellationToken.None" />. </param>
+        /// <returns> A collection of location data that may take multiple service requests to iterate over. </returns>
+        /// <exception cref="InvalidOperationException"> <paramref name="subscriptionId"/> is null. </exception>
+        public Pageable<LocationData> List(string subscriptionId = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (string.IsNullOrWhiteSpace(subscriptionId))
+            {
+                subscriptionId = GetDefaultSubscription(cancellationToken).ConfigureAwait(false).GetAwaiter().GetResult();
+                if (subscriptionId == null)
+                {
+                    throw new InvalidOperationException("Please select a default subscription");
+                }
+            }
+
+            return new PhWrappingPageable<Azure.ResourceManager.Resources.Models.Location, LocationData>(SubscriptionsClient.ListLocations(subscriptionId, cancellationToken), s => s.DisplayName);
+        }
+
+        /// <summary>
+        /// Lists all geo-locations.
+        /// </summary>
+        /// <param name="subscriptionId"> The Id of the target subscription. </param>
+        /// <param name="token"> A token to allow the caller to cancel the call to the service. The default value is <see cref="P:System.Threading.CancellationToken.None" />. </param>
+        /// <returns> An async collection of location data that may take multiple service requests to iterate over. </returns>
+        /// <exception cref="InvalidOperationException"> <paramref name="subscriptionId"/> is null. </exception>
+        public AsyncPageable<LocationData> ListAsync(string subscriptionId = null, CancellationToken token = default(CancellationToken))
```

Comment:
should not take in a subscription Id, use the Id.Subscription property of the class

Reply:
still needs addressed

Reply:
still need to remove this